identity/backend/src/main/java/io/camunda/identity/usermanagement/service/UserService.java

```diff
@@ -40,64 +45,66 @@ public UserService(
     this.passwordEncoder = passwordEncoder;
   }
 
-  public CamundaUser createUser(final CamundaUserWithPassword userWithCredential) {
+  public CamundaUser createUserFailIfExists(
+      @Valid final CamundaUserWithPassword camundaUserWithPassword) {
     try {
-      final UserDetails userDetails =
-          User.withUsername(userWithCredential.getUsername())
-              .password(userWithCredential.getPassword())
-              .passwordEncoder(passwordEncoder::encode)
-              .disabled(!userWithCredential.isEnabled())
-              .roles("DEFAULT_USER")
-              .build();
-      camundaUserDetailsManager.createUser(userDetails);
-      final var createdUser = userProfileRepository.findByUsername(userDetails.getUsername());
-      userProfileRepository.save(
-          new Profile(
-              createdUser.getId(), userWithCredential.getEmail(), userWithCredential.getName()));
-      return userProfileRepository.findByUsername(userWithCredential.getUsername());
+      return createUser(camundaUserWithPassword);
     } catch (final DuplicateKeyException e) {
-      throw new RuntimeException("user.duplicate");
+      throw new IllegalArgumentException("User invalid.");
```

Comment:
❓ I'm never too sure on these exception messages with how much we want to expose but here we know that the issue is actually that the user (or rather the unique key) already exists, I think `User already exists.` may be more suitable here